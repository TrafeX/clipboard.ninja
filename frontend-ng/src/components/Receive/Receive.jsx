import {
  Box,
  Code,
  Flex,
  List,
  ListIcon,
  ListItem
} from "@chakra-ui/react";
import { IoReturnDownForwardSharp } from "react-icons/io5";

const Receive = ({ receivedMessages }) => (
  <Box>
    <List>
      {receivedMessages &&
        receivedMessages.map((message, i) => {
          return (
            <ListItem key={i}>
              <Flex direction="row">
                <ListIcon as={IoReturnDownForwardSharp} color={"brand.600"} />
                <Code style={{whiteSpace: "pre-wrap"}}>{message}</Code>
              </Flex>
            </ListItem>
          );
        })}
    </List>
  </Box>
);

export default Receive;
