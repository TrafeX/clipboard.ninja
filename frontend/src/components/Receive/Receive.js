// @flow
import React from 'react';

const style = {
  ul: {
    textAlign: 'left',
  }
};

type Props = {
  receivedMessages: Array<string>,
}

const Receive = ({receivedMessages}: Props) => (
  <div>
    <ul style={style.ul}>
    { receivedMessages && receivedMessages.map((message, i) => {
      return <li key={i}><pre>{message}</pre></li>
    }) }
    </ul>
  </div>
);

export default Receive;
