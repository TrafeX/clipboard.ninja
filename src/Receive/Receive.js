// @flow
import React from 'react';

const style = {
  ul: {
    textAlign: 'left',
  }
};

type Props = {
  receivedMessages: Array<string>,
}

const Receive = (props: Props) => (
  <div>
    <ul style={style.ul}>
    { props.receivedMessages.map((message, i) => {
      return <li key={i}><pre>{message}</pre></li>
    }) }
    </ul>
  </div>
);

export default Receive;
