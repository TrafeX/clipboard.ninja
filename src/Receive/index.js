import Receive from './Receive';

export default Receive;
