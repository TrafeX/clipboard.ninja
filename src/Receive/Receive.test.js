import React from 'react';
import ReactDOM from 'react-dom';
import Receive from './Receive';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Receive
      receivedMessage='Hello world!'
    />, div);
});
