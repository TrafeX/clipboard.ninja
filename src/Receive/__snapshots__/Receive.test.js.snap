// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Receive
    receivedMessages={
      Array [
        "Hello",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        <li>
          <pre>
            Hello
          </pre>
        </li>
      </ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <pre>
              Hello
            </pre>
          </li>,
        ],
        "style": Object {
          "textAlign": "left",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <pre>
              Hello
            </pre>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hello",
            },
            "ref": null,
            "rendered": "Hello",
            "type": "pre",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          style={
            Object {
              "textAlign": "left",
            }
          }
        >
          <li>
            <pre>
              Hello
            </pre>
          </li>
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <pre>
                Hello
              </pre>
            </li>,
          ],
          "style": Object {
            "textAlign": "left",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": <pre>
                Hello
              </pre>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hello",
              },
              "ref": null,
              "rendered": "Hello",
              "type": "pre",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
